pkg/lib/cockpit-components-file-autocomplete.jsx

```diff
@@ -56,16 +63,28 @@ var FileAutoComplete = React.createClass({
     },
 
     onChange: function(value) {
-        if (value && value.indexOf("/") !== 0)
+        if (value && value.indexOf("/") !== 0) {
             value = "/" + value;
+        }
 
-        if (!this.updateIfDirectoryChanged(value));
-            this.filterFiles(value);
-
-        this.setState({
+        var stateUpdate = {
             value: value,
+        };
+
+        if (!this.updateIfDirectoryChanged(value)) {
+            Object.assign(stateUpdate, this.filterFiles(value));
```

Comment:
Can we stick with the more generic setState?

Reply:
I need this so onChange callback is not called twice and value/error pair stays consistent. Also filterFiles is used only in this function so its meaning is code separation and it is enough to call setState only once.